api/src/database/helpers/schema/dialects/mysql.ts

```diff
@@ -70,4 +70,33 @@ export class SchemaHelperMySQL extends SchemaHelper {
 			groupByFields.push(...sortRecords.map(({ alias }) => alias));
 		}
 	}
+
+	override async createIndex(
+		collection: string,
+		field: string,
+		options: CreateIndexOptions = {},
+	): Promise<Knex.SchemaBuilder> {
+		const isUnique = Boolean(options.unique);
+		const constraintName = this.generateIndexName(isUnique ? 'unique' : 'index', collection, field);
+
+		if (options.tryNonBlocking) {
+			/*
+			Seems it is not possible to determine whether "ALGORITHM=INPLACE LOCK=NONE" will be supported
+			so we're just going to send it and fall back to blocking index creation on error
+			
+			https://dev.mysql.com/doc/refman/8.4/en/create-index.html#:~:text=engine%20is%20changed.-,Table%20Copying%20and%20Locking%20Options,-ALGORITHM%20and%20LOCK
+			*/
+			return this.knex
+				.raw(`CREATE ${isUnique ? 'UNIQUE ' : ''}INDEX ?? ON ?? (??) ALGORITHM=INPLACE LOCK=NONE`, [
+					constraintName,
+					collection,
+					field,
+				])
+				.catch(() =>
+					this.knex.raw(`CREATE ${isUnique ? 'UNIQUE ' : ''}INDEX ?? ON ?? (??)`, [constraintName, collection, field]),
+				);
+		}
+
+		return this.knex.raw(`CREATE ${isUnique ? 'UNIQUE ' : ''}INDEX ?? ON ?? (??)`, [constraintName, collection, field]);
+	}
```

Comment:
We should be able to simplify this to a try/catch or a single .catch, the last raw query looks to be redundant?

```suggestion
		/*
			Seems it is not possible to determine whether "ALGORITHM=INPLACE LOCK=NONE" will be supported
			so we're just going to send it and fall back to blocking index creation on error

			https://dev.mysql.com/doc/refman/8.4/en/create-index.html#:~:text=engine%20is%20changed.-,Table%20Copying%20and%20Locking%20Options,-ALGORITHM%20and%20LOCK
			*/
		try {
			if (options.tryNonBlocking) {
				return this.knex.raw(`CREATE ${isUnique ? 'UNIQUE ' : ''}INDEX ?? ON ?? (??) ALGORITHM=INPLACE LOCK=NONE`, [
					constraintName,
					collection,
					field,
				]);
			}
		} catch {
			return this.knex.raw(`CREATE ${isUnique ? 'UNIQUE ' : ''}INDEX ?? ON ?? (??)`, [
				constraintName,
				collection,
				field,
			]);
		}
```